timemachine/cpp/src/device_buffer.cu

```diff
@@ -0,0 +1,20 @@
+#include "device_buffer.hpp"
+#include "gpu_utils.cuh"
+#include <cstddef>
+
+namespace timemachine {
+
+template <typename T>
+DeviceBuffer<T>::DeviceBuffer(const std::size_t length) : size(length * sizeof(T)), data(allocate_(size)) {}
+
+template <typename T> DeviceBuffer<T>::~DeviceBuffer() { gpuErrchk(cudaFree(data)); }
+
+template <typename T> T *DeviceBuffer<T>::allocate_(const std::size_t size) {
+    T *data;
```

Comment:
Should this be a different name, perhaps `buffer`? It is confusing that data is both a private member and a free variable in this context.

Reply:
Agreed, fixed in 6bcef8b.